server.go

```diff
@@ -1138,9 +1160,12 @@ func (s *server) InboundPeerConnected(conn net.Conn) {
 		// peer to the peer garbage collection goroutine.
 		srvrLog.Debugf("Disconnecting stale connection to %v",
 			connectedPeer)
-		connectedPeer.Disconnect(errors.New("remove stale connection"))
 
+		// Remove the current peer from the server's internal state and
+		// signal that the peer termination watcher does not need to
+		// execute for this peer.
 		s.removePeer(connectedPeer)
+		s.ignorePeerTermination[connectedPeer] = struct{}{}
```

Comment:
Maybe add a default case for unexpected errors (if more errors from the method can happen sometime in the future).

Reply:
I replaced the switch with a scoped if stmt, should be cleaner and no need to handle a default case :)